modules/core/src/controllers/controller.ts

```diff
@@ -44,12 +44,43 @@ const EVENT_TYPES = {
   KEYBOARD: ['keydown']
 };
 
+type ControllerProps = Record<string, any>;
+
 export default class Controller {
-  constructor(ControllerState, options = {}) {
+  ControllerState;
+  controllerState: Record<string, any> | null = null;
+  controllerStateProps: Record<string, any> | null = null;
+  transitionManager: TransitionManager;
+  _transition;
+  _events: Record<string, any> | null = null;
+  eventManager: any = null;
+  _interactionState = {
+    isDragging: false
+  };
+  _customEvents = [];
+  onViewStateChange = null;
+  onStateChange = null;
+
+  makeViewport;
+  _eventStartBlocked;
+  _state;
+
+  _panMove;
+  invertPan;
+  dragMode;
+
+  inertia: number = 0;
+
+  scrollZoom: boolean = true;
+  dragPan: boolean = true;
+  dragRotate: boolean = true;
+  doubleClickZoom: boolean = true;
+  touchZoom: boolean = true;
+  touchRotate: boolean = false;
+  keyboard: boolean = true;
```

Comment:
Some of these can be objects, see https://deck.gl/docs/api-reference/core/controller#options

Reply:
> Some of these can be objects

Fixed